src/jdk.incubator.vector/share/classes/jdk/incubator/vector/VectorMathLibrary.java

```diff
@@ -101,31 +174,51 @@ static String precisionLevel(Operator op) {
         //              "sve/advsimd" for sve/neon implementations
         //     e.g. sinfx_u10sve is the method for computing vector float sin using SVE instructions
         //          cosd2_u10advsimd is the method for computing 2 elements vector double cos using NEON instructions
-        static String symbolName(Operator op, VectorSpecies<?> vspecies) {
+        @Override
+        public String symbolName(Operator op, VectorSpecies<?> vspecies) {
             return String.format("%s%s%d_%s%s", op.operatorName(),
                                  (vspecies.elementType() == float.class ? "f" : "d"),
                                  vspecies.length(),
                                  precisionLevel(op),
                                  suffix(vspecies.vectorShape()));
         }
+
+        @Override
+        public boolean isSupported(Operator op, VectorSpecies<?> vspecies) {
+            Class<?> etype = vspecies.elementType();
+            if (etype != float.class && etype != double.class) {
+                return false; // only FP element types are supported
+            }
+            int maxLaneCount = VectorSupport.getMaxLaneCount(vspecies.elementType());
+            if (vspecies.length() > maxLaneCount) {
+                return false; // lacking vector support
+            }
+            if (vspecies.vectorBitSize() < 128) {
+                return false; // 64-bit vectors are not supported
+            }
```

Comment:
Thanks for your refactor. It's really a good job!  

It seems float type support 64-bit vector operations before (see https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp#L9835). Will this change the behavior of 64-bit float vector? Thanks!

Reply:
Thanks for taking a look. 

In the latest version, Float64Vector cases should be properly handled.

Reply:
Oh, yes. That's right! Thanks and sorry for my distraction.